pkg/manifests/manifests_test.go

```diff
@@ -1188,6 +1164,86 @@ func TestPrometheusK8sRemoteWriteURLs(t *testing.T) {
 	}
 }
 
+func TestRemoteWriteAuthorizationConfig(t *testing.T) {
+	for _, tc := range []struct {
+		name    string
+		config  string
+		checkFn func(*testing.T, monv1.RemoteWriteSpec)
+	}{
+		{
+			name: "basic authentication configuration",
+			config: `prometheusK8s:
+  remoteWrite:
+  - url: "https://basicAuth.remotewrite.com/api/write"
+    basicAuth:
+      username:
+        name: remoteWriteAuth
+        key: user
+      password:
+        name: remoteWriteAuth
+        key: password
+`,
+			checkFn: func(t *testing.T, target monv1.RemoteWriteSpec) {
+				if target.BasicAuth.Username.Name != "remoteWriteAuth" && target.BasicAuth.Username.Key != "user" {
+					t.Fatal("Username section not correct in RemoteWriteSpec.BasicAuth")
+				}
```

Comment:
In general, I like failing tests to report `expected <this>, got <that>`. It makes easier to see what didn't match.

```suggestion
					t.Fatalf("expecting username %s, got %s", target.BasicAuth.Username)
```
```suggestion
				if target.BasicAuth.Username.Name != "remoteWriteAuth" {
					t.Fatalf("expecting BasicAuth username secret 'remoteWriteAuth', got %q", target.BasicAuth.Username)
				}
				if target.BasicAuth.Username.Key != "user" {
					t.Fatalf("expecting BasicAuth username key 'user', got %q", target.BasicAuth.Key)
				}
```